block/production_test.go

```diff
@@ -0,0 +1,158 @@
+	require.Equal(initialHeight, manager.store.Height())
+
+	mCtx, cancel := context.WithTimeout(context.Background(), runTime)
+	defer cancel()
+	go manager.ProduceBlockLoop(mCtx)
+	go managerWithEmptyBlocks.ProduceBlockLoop(mCtx)
+	<-mCtx.Done()
+
+	require.Greater(manager.store.Height(), initialHeight)
+	require.Greater(managerWithEmptyBlocks.store.Height(), initialHeight)
+	assert.Greater(managerWithEmptyBlocks.store.Height(), manager.store.Height())
+
+	// Check that blocks are created with empty blocks feature disabled
+	assert.LessOrEqual(manager.store.Height(), uint64(runTime/EmptyBlocksMaxTime))
+	assert.LessOrEqual(managerWithEmptyBlocks.store.Height(), uint64(runTime/blockTime))
+
+	for i := uint64(2); i < managerWithEmptyBlocks.store.Height(); i++ {
+		prevBlock, err := managerWithEmptyBlocks.store.LoadBlock(i - 1)
+		assert.NoError(err)
+
+		block, err := managerWithEmptyBlocks.store.LoadBlock(i)
+		assert.NoError(err)
+		assert.NotZero(block.Header.Time)
+
+		diff := time.Unix(0, int64(block.Header.Time)).Sub(time.Unix(0, int64(prevBlock.Header.Time)))
+		assert.Greater(diff, blockTime-blockTime/10)
+		assert.Less(diff, blockTime+blockTime/10)
+	}
+
+	for i := uint64(2); i < manager.store.Height(); i++ {
+		prevBlock, err := manager.store.LoadBlock(i - 1)
+		assert.NoError(err)
+
+		block, err := manager.store.LoadBlock(i)
+		assert.NoError(err)
+		assert.NotZero(block.Header.Time)
+
+		diff := time.Unix(0, int64(block.Header.Time)).Sub(time.Unix(0, int64(prevBlock.Header.Time)))
+		assert.Greater(diff, manager.conf.EmptyBlocksMaxTime)
+	}
+}
+
+func TestCreateEmptyBlocksNew(t *testing.T) {
+	t.Skip("FIXME: fails to submit tx to test the empty blocks feature")
+	assert := assert.New(t)
+	require := require.New(t)
+	app := testutil.GetAppMock()
+	// Create proxy app
+	clientCreator := proxy.NewLocalClientCreator(app)
+	proxyApp := proxy.NewAppConns(clientCreator)
+	err := proxyApp.Start()
+	require.NoError(err)
+	// Init manager
+	managerConfig := getManagerConfig()
+	managerConfig.BlockTime = 200 * time.Millisecond
+	managerConfig.EmptyBlocksMaxTime = 1 * time.Second
+	manager, err := getManager(managerConfig, nil, nil, 1, 1, 0, proxyApp, nil)
+	require.NoError(err)
+
+	abciClient, err := clientCreator.NewABCIClient()
+	require.NoError(err)
+	require.NotNil(abciClient)
+	require.NoError(abciClient.Start())
+	defer abciClient.Stop()
+
+	mempoolCfg := tmcfg.DefaultMempoolConfig()
+	mempoolCfg.KeepInvalidTxsInCache = false
+	mempoolCfg.Recheck = false
+
+	mpool := mempoolv1.NewTxMempool(log.TestingLogger(), tmcfg.DefaultMempoolConfig(), proxy.NewAppConnMempool(abciClient), 0)
+
+	//Check initial height
+	expectedHeight := uint64(0)
+	assert.Equal(expectedHeight, manager.store.Height())
+
+	mCtx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+	go manager.ProduceBlockLoop(mCtx)
+
+	<-time.Tick(1 * time.Second)
+	err = mpool.CheckTx([]byte{1, 2, 3, 4}, nil, mempool.TxInfo{})
```

Comment:
@omritoptix If you can take a look
I can't manage to send a tx

all the blocks produced are empty
The test is skipped for now. I've tested it manually

Reply:
Will do.